akka-bbb-apps/src/main/scala/org/bigbluebutton/core/models/Plugins.scala

```diff
@@ -99,6 +112,96 @@ object PluginModel {
+    val updatedPluginSetting: Map[String, ClientSettings.Plugin] = currentPluginSettings.get(pluginName) match {
+      // Plugin exists: add setting in its existing settings map
+      case Some(
+        pluginSettings: ClientSettings.Plugin
+      ) => Map(pluginName -> ClientSettings.Plugin(pluginName, pluginSettings.settings + (settingKey -> settingValue)))
+      // Plugin not found: create a new plugin entry with the given setting
+      case None => Map(pluginName -> ClientSettings.Plugin(pluginName, Map(settingKey -> settingValue)))
+    }
+    currentPluginSettings ++ updatedPluginSetting
+  }
+
+  private def validateAndApplySettingHelper(
+                                       pluginSettingsMap: Map[String, ClientSettings.Plugin],
+                                       pluginName: String,
+                                       schema: PluginSettingSchema
+                                     ): (Boolean, Map[String, ClientSettings.Plugin]) = {
+    val settingValueOpt = ClientSettings.getPluginSettingValue(pluginSettingsMap, pluginName, schema.name)
+    val valueIsCorrectType = checkSettingsType(schema.`type`, settingValueOpt)
+
+    settingValueOpt match {
+      case Some(value) =>
+        if (!valueIsCorrectType) {
+          logger.error("Plugin [{}]: Setting [{}] has a value [{}] of incorrect type. Expected [{}]. Plugin will not be loaded.",
+            pluginName, schema.name, value, schema.`type`)
+        }
+        (valueIsCorrectType, pluginSettingsMap)
+
+      case None =>
+        val defaultValid = checkSettingsType(schema.`type`, Option(schema.defaultValue))
+        if (defaultValid) {
+          logger.warn("Plugin [{}]: Required setting [{}] not found. Falling back to default value [{}]",
+            pluginName, schema.name, schema.defaultValue)
+
+          val updatedpluginSettingMap = addPluginSettingEntry(pluginSettingsMap, pluginName, schema.name, schema.defaultValue)
+          (true, updatedpluginSettingMap)
+        } else {
+          logger.error("Plugin [{}]: Required setting [{}] is missing and default value [{}] does not match expected type [{}]",
+            pluginName, schema.name, schema.defaultValue, schema.`type`)
+          (false, pluginSettingsMap)
+        }
+    }
+  }
+
+  private def validatePluginsBeforeCreateModel(
+                                                instance: PluginModel,
+                                                clientSettings: Map[String, Object]
+                                              ): (PluginModel, List[ClientSettings.Plugin]) = {
+
+    var pluginSettings = getPluginsFromConfig(clientSettings)
+
+    instance.plugins = instance.plugins.filter { case (_, plugin) =>
+      val pluginName = plugin.manifest.content.name
+      logger.info("Validating settings for plugin {}", pluginName)
+
+      plugin.manifest.content.settingsSchema match {
+        case Some(schemaList) =>
+          schemaList.forall { schema =>
```

Comment:
I had a hard time understanding what "schema" meant, in other parts of the code.
Please rename it to `settingSchema` or simply `setting`, as just "schema" is too vague.